docs/design/basic_gateway/design.md

```diff
@@ -0,0 +1,411 @@
+## On-demand installing of table entries based on basic gateways
+
+### 1. Requirements
+
+In a cloud network, rules need to be installed on the computing nodes when the virtual machine is online. If the scale of the VPC is large, it takes a long time to get VMs online. In order to increase the speed of the virtual machine going online, it is necessary to install the rules on the computing nodes as needed. It needs to satisfy:
+
+-   Only a small number of necessary rules are installed when the virtual machine goes online, and the remaining rules are gradually installed as needed after the virtual machine is online；
+
+-   The size of the rules installed when virtual machine is online is decoupled from the VPC scale；
+
+-   The number of rules deployed on the compute node is greatly reduced, reducing the risk of overload and failure of high-density servers；
+
+### 2. Gateway node forwarding model
+
+In order to reduce the number of rules installed when the virtual machine is online and increase the speed of the virtual machine going online, we designe a gateway node forwarding model. Each gateway node stores the global rule information of one VPC or multiple VPCs. In the gateway node forwarding model, when a virtual machine is online, the rules configuration is mainly divided into two parts:
+
+-   Installing the configuration related to the virtual machine to access the network, and configure the bucket-to-gateway mapping on the host-side OVS   (buckets are described below). At this point, the virtual machine has been successfully launched.
+
+-    When the newly-launched virtual machine communicates with other virtual machines, it needs to query the rules in the OVS flow table on the computing node. Since the newly-launched virtual machine only matches the default rules for forwarding to the gateway node, the first packet of the flow will be sent to the gateway node. Then, the gateway node queries the global rules and installs the direct path rules for the OVS in the compute node  where the virtual machine is located. Then the virtual machine pairs can use direct path forwarding if the rules have been installed.
```

Comment:
Are you going to query rules in the OVS flow table? How will you do that without buffering packets? Do you mean OVS matches existing rules?

Can you explain more what do you mean by: "the gateway node queries the global rules and installs the direct path rules for the OVS"?